staging/publishing/rules.yaml

```diff
@@ -7,25 +7,25 @@ rules:
       dirs:
       - staging/src/k8s.io/apimachinery
   - name: release-1.28
```

Comment:
change to go 1.23.6 looks fine, I'm surprised we still have release-1.28 branches present in this file

Reply:
should we drop the 1.28 branch stanzas first, since we didn't actually bump 1.28 to go 1.23?

Reply:
we should probably remove 1.28 totally from this file? ( https://endoflife.date/kubernetes )

Reply:
i would say that, i will drop 1.28